packages/salesforcedx-vscode-core/src/commands/templates/internalCommandUtils.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer,
+  PostconditionChecker,
+  PreconditionChecker
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { Uri } from 'vscode';
+import { channelService } from '../../channels';
+import { notificationService } from '../../notifications';
+import { sfdxCoreSettings } from '../../settings';
+import { telemetryService } from '../../telemetry';
+import { AURA_METADATA_TYPE, LWC_METADATA_TYPE } from './metadataTypeConstants';
+
+export class InternalDevWorkspaceChecker implements PreconditionChecker {
+  public check(): boolean {
+    return sfdxCoreSettings.getInternalDev();
+  }
+}
+
+export class FileInternalPathGatherer
+  implements ParametersGatherer<{ outputdir: string }> {
+  private filePath: string;
+  public constructor(uri: Uri) {
+    // add some validations here.
+    this.filePath = uri.fsPath;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ outputdir: string }>
+  > {
+    const outputdir = this.filePath;
+    return outputdir
+      ? { type: 'CONTINUE', data: { outputdir } }
+      : { type: 'CANCEL' };
+  }
+}
+
+export class InternalSourcePathChecker
```

Comment:
Validates that the file path is the correct one based on the type of metadata that the user is trying to create.

Reply:
I don't think we should enforce this

Reply:
Why not? This is based on the source path conversation we had earlier, where `aura` metadata should only be created in source paths that contain `/components/{namespace}` and `/modules/{namespace}` for the `lwc` scenarios.

Reply:
This is true for a majority of projects in core, but lwc can be configured to read modules from any directory. I just don't see any value in limiting where internal devs/ off-core devs can create these components, especially now that you have it working with a right click action on a folder.